src/sources/static_metrics.rs

```diff
@@ -0,0 +1,287 @@
+use std::collections::BTreeMap;
+use std::num::NonZero;
+use std::time::Duration;
+
+use chrono::Utc;
+use futures::StreamExt;
+use serde_with::serde_as;
+use tokio::time;
+use tokio_stream::wrappers::IntervalStream;
+use vector_lib::configurable::configurable_component;
+use vector_lib::internal_event::{CountByteSize, InternalEventHandle as _};
+use vector_lib::{config::LogNamespace, ByteSizeOf, EstimatedJsonEncodedSizeOf};
+
+use crate::{
+    config::{SourceConfig, SourceContext, SourceOutput},
+    event::{
+        metric::{MetricData, MetricName, MetricSeries, MetricTime, MetricValue},
+        EventMetadata, Metric, MetricKind,
+    },
+    internal_events::{EventsReceived, StaticMetricsBytesReceived, StreamClosedError},
+    shutdown::ShutdownSignal,
+    SourceSender,
+};
+
+/// Configuration for the `static_metrics` source.
+#[serde_as]
+#[configurable_component(source(
+    "static_metrics",
+    "Produce static metrics defined in configuration."
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct StaticMetricsConfig {
+    /// The interval between metric emitting, in seconds.
+    #[serde_as(as = "serde_with::DurationSecondsWithFrac<f64>")]
+    #[serde(default = "default_interval")]
+    #[configurable(metadata(docs::human_name = "Emitting interval"))]
+    pub interval_secs: Duration,
+
+    /// Overrides the default namespace for the metrics emitted by the source.
+    #[serde(default = "default_namespace")]
+    pub namespace: String,
+
+    #[configurable(derived)]
+    #[serde(default = "default_metrics")]
+    pub metrics: Vec<StaticMetricConfig>,
+}
+
+impl Default for StaticMetricsConfig {
+    fn default() -> Self {
+        Self {
+            interval_secs: default_interval(),
+            metrics: Vec::default(),
+            namespace: default_namespace(),
+        }
+    }
+}
+
+/// Tag configuration for the `internal_metrics` source.
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
```

Comment:
I don't if its intuitive to use `#[derive(Default)] here...

This will use name="" and value=0.0, I think its too error-prone for the end user. 

I think we should make the `name` and `value` attributes mandatory. The tags could default as an empty map using the `#[serde(default)] attribute

Reply:
Don't worry! it was just a suggestion 😄